ci-operator/step-registry/baremetal/lab/agent/day2/baremetal-lab-agent-day2-commands.sh

```diff
@@ -0,0 +1,181 @@
+  local ret=$?
+  if [ $ret -ne 0 ]; then
+    echo "Failed to mount the ISO image in #${host} via virtual media."
+    touch /tmp/virtual_media_mount_failure
+    return 1
+  fi
+  return 0
+}
+
+function get_ready_nodes_count() {
+  oc get nodes \
+    -o jsonpath='{range .items[*]}{.metadata.name}{","}{.status.conditions[?(@.type=="Ready")].status}{"\n"}{end}' | \
+    grep -c -E ",True$"
+}
+
+function approve_csrs() {
+  while [[ ! -f '/tmp/agent-day2-complete' ]]; do
+    echo "approve_csrs() running..."
+    oc get csr -o go-template='{{range .items}}{{if not .status}}{{.metadata.name}}{{"\n"}}{{end}}{{end}}' \
+      | xargs --no-run-if-empty oc adm certificate approve || true
+    sleep 30
+  done
+}
+
+function monitor_workers()
+{
+  local DAY2_IPS=""
+  expected_nodes=$(( $(get_ready_nodes_count) + ADDITIONAL_WORKERS ))
+  # shellcheck disable=SC2154
+  for bmhost in $(yq e -o=j -I=0 '.[] | select(.name|test("-a-"))' "${SHARED_DIR}/hosts.yaml"); do
+    # shellcheck disable=SC1090
+    . <(echo "$bmhost" | yq e 'to_entries | .[] | (.key + "=\"" + .value + "\"")')
+    DAY2_IPS+="${ip},"
+  done
+  DAY2_IPS=${DAY2_IPS%,}
+  /tmp/oc adm node-image monitor --ip-addresses "${DAY2_IPS}" -a "${day2_pull_secret}" --insecure=true
+  ready_nodes=$(get_ready_nodes_count)
+  if [ "${ready_nodes}" != "${expected_nodes}" ]; then
+    echo "Day2 install fail..."
+    exit 1
+  fi
+  touch /tmp/agent-day2-complete
+  return 0
+}
+
+
+SSHOPTS=(-o 'ConnectTimeout=5'
+  -o 'StrictHostKeyChecking=no'
+  -o 'UserKnownHostsFile=/dev/null'
+  -o 'ServerAliveInterval=90'
+  -o LogLevel=ERROR
+  -i "${CLUSTER_PROFILE_DIR}/ssh-key")
+
+
+export KUBECONFIG="$SHARED_DIR/kubeconfig"
+export http_proxy="${proxy}" https_proxy="${proxy}" HTTP_PROXY="${proxy}" HTTPS_PROXY="${proxy}"
+
+day2_pull_secret="${SHARED_DIR}/day2_pull_secret"
+cat "${CLUSTER_PROFILE_DIR}/pull-secret" > "${day2_pull_secret}"
+
+echo "Extract the latest oc client..."
+oc adm release extract -a "${day2_pull_secret}" "${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}" \
+   --command=oc --to=/tmp --insecure=true
+
+/tmp/oc version
+/cli/oc version
+
+if [ "${DISCONNECTED}" == "true" ] && [ -f "${SHARED_DIR}/install-config-mirror.yaml.patch" ]; then
+  OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE="$(<"${CLUSTER_PROFILE_DIR}/mirror_registry_url")/${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE#*/}"
+  # use registry credential for disconnected auth
+  oc get secret -n openshift-config pull-secret -o jsonpath='{.data.\.dockerconfigjson}' | base64 -d > "${day2_pull_secret}"
+fi
+
+DAY2_INSTALL_DIR="${DAY2_INSTALL_DIR:-/tmp/installer_day2}"
+mkdir -p "${DAY2_INSTALL_DIR}"
+cp "${SHARED_DIR}/nodes-config.yaml" "${DAY2_INSTALL_DIR}/"
+cp "${SHARED_DIR}/nodes-config.yaml" "${ARTIFACT_DIR}/"
+
+echo "Create node.iso for day2 worker nodes..."
+/tmp/oc adm node-image create --dir="${DAY2_INSTALL_DIR}" -a "${day2_pull_secret}" --insecure=true
```

Comment:
Please check whether this command can execute without `-a "${day2_pull_secret}"` option, since `export KUBECONFIG` is already set."